### DISCUSSION TITLE
PowerDNS

### DISCUSSION THREAD

**Discussion by TheTechromancer** [blacklanternsecurity/bbot#2945] [Module Requests]
PowerDNS is an open source DNS server and resolver written in C++, that we could possibly integrate into BBOT. This would allow us to improve our DNS accuracy and speed up resolution. 

https://github.com/PowerDNS/pdns

https://youtu.be/120iDYzscD4?si=khKKAU7fieSNaDS6&t=750

@Sh4d0wHunt3rX @liquidsec

**Comment by TheTechromancer** [comment]
Even if we don't end up packaging this directly into bbot, it might work well as a sidecar deployment, using it as the primary DNS server for resolution + brute forcing.

**Comment by liquidsec** [comment]
Are you concerned about going so fast we get blocked or limited by the upstream servers it would talk to?

**Comment by TheTechromancer** [comment]
Possibly yeah. In the defcon talk in that YouTube video, I think he set up multiple resolvers and load balanced between them. But he was doing an insane volume of queries so we might not need to. We would need to test it.
